src/events.cpp

```diff
@@ -36,15 +36,17 @@ void RegisterEventListeners()
 {
 	FOR_EACH_VEC(g_vecEventListeners, i)
 	{
-		g_gameEventManager->AddListener(g_vecEventListeners[i], g_vecEventListeners[i]->GetEventName(), true);
+		if (g_gameEventManager)
+			g_gameEventManager->AddListener(g_vecEventListeners[i], g_vecEventListeners[i]->GetEventName(), true);
 	}
 }
 
 void UnregisterEventListeners()
 {
 	FOR_EACH_VEC(g_vecEventListeners, i)
 	{
-		g_gameEventManager->RemoveListener(g_vecEventListeners[i]);
+		if (g_gameEventManager)
+			g_gameEventManager->RemoveListener(g_vecEventListeners[i]);
```

Comment:
If you are gonna nullcheck `g_gameEventManager` here, better just do it once at the beginning and early return if null.

Reply:
good point